src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -238,6 +241,22 @@ static class OptionsValues {
     }
 
     public static final String OPTIONS_RESOURCE = "jdk/tools/jlink/internal/options";
+    // Release information as in the java.base module for target image
+    public static final String JDK_RELEASE_RESOURCE = "jdk/internal/jmod/resources/release.txt";
+
+    /**
+     * Read the release.txt from the module.
+     */
+    private static Optional<String> getReleaseInfo(ModuleReference mref) throws IOException {
+        var in = mref.open().open(JDK_RELEASE_RESOURCE);
+        if (in.isEmpty()) {
+            return Optional.empty();
+        }
+
+        try (var r = new BufferedReader(new InputStreamReader(in.get()))) {
```

Comment:
It's not immediately obvious that `in` is an `Optional<InputStream>` , so maybe rename and/or avoid var for this case.